errors/context.go

```diff
@@ -0,0 +1,193 @@
+package errors
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/mailgun/holster/v5/callstack"
+	"github.com/sirupsen/logrus"
+)
+
+// HasContext Implement this interface to pass along unstructured context to the logger
```

Comment:
When you see word Context you immediately assume `context.Context`, so I suggest changing `Context` to `Fields` in this context, the pun is not intended )